### PR TITLE
Use reths EthApiError instead of ours

### LINKED ISSUE #393: Replace our eth api errors with reth's
We need to compare rpc errors with reth and fix them accordingly

### PR DESCRIPTION
**Pull request by ercecan**
# Description
Uses reth's EthApiError in functiosn and rpcs instead of ours

## Linked Issues
- Fixes #393

### CODE REVIEW

File: crates/evm/src/evm/tests.rs [hunk 1]
```diff
@@ -33,7 +32,7 @@ fn simple_contract_execution_sov_state() {
     simple_contract_execution(evm_db);
 }
 
-fn simple_contract_execution<DB: Database<Error = Infallible> + DatabaseCommit + InitEvmDb>(
+fn simple_contract_execution<DB: Database<Error = DBError> + DatabaseCommit + InitEvmDb>(
```
Comment (kpp) [inline crates/evm/src/evm/tests.rs:35]: Actually no need to specify `<Error =..>`

**Comment by kpp** [comment]
Very good

File: crates/evm/src/evm/tests.rs (hunk 1 above)
Comment (ercecan) [reply crates/evm/src/evm/tests.rs:35]: I think we need to do, it threw this error
```
note: trait defined here, with 0 generic parameters
sov-demo-prover-guest-mock:   --> /Users/erce/.cargo/registry/src/index.crates.io-6f17d22bba15001f/revm-primitives-3.1.0/src/db.rs:11:11
sov-demo-prover-guest-mock:    |
sov-demo-prover-guest-mock: 11 | pub trait Database {
sov-demo-prover-guest-mock:    |           ^^^^^^^^
sov-demo-prover-guest-mock: help: replace the generic bound with the associated type
sov-demo-prover-guest-mock:    |
sov-demo-prover-guest-mock: 78 |     DB: Database<Error = DBError> + DatabaseCommit,
```

**Review by eyusufatik — APPROVED** [review]
looks good

**Pull request closed without merging**